### PR TITLE
chore: add proposal for MUIv5 migration

### PR DESCRIPTION
**Pull request by ozgurkececioglu**
### Description

Proposal for Material-UI v5 migration plan

#### When to approve? ✅

**You are OK** with merging this PR and

1. You understand the plan and you don't have any objection to it.
2. You have optional requests related to planning or the goal.

### CODE REVIEW

File: proposals/mui-v5-migration.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Description
+
+This document lists the goal, plan and the decisions related to Material UI migration from v4 to v5.
```
Comment (dmaklygin) [inline proposals/mui-v5-migration.md:3]: `decisions` might be removed, they are not listed here. wdyt?

File: proposals/mui-v5-migration.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+As a core foundation, we are using MUI for several reasons like theming and JSS solutions. So, it would be good to migrate it to version 5 to be able to have more reliable and supported base with better CSS-in-JS performance.
+
+# Goal
+
+The goal of this migration is to have fully migrated core components of MUI from version 5 and emotion based CSS-in-JS solution with minimal number of breaking changes reflected to Picasso clients. It also means that we shouldn't be reflecting anything related to MUI did internally for meeting their own design purposes.
+
+To be able to make things straight forward, it would be good to postpone removal of all JSS parts (like @mui/styles package usages) from the repo before everthing is working fine again. So, `styled` or `TSS-react` solutions can be discussed in further iterations.
```
Comment (dmaklygin) [inline proposals/mui-v5-migration.md:11]: `everthing` -> `everything`

**Review by TomasSlama — COMMENTED** [review]
I really like the option, that we would be able to update components and merge to master, my only concern is, if we keep two roots, will it not affect badly performance?, like bundle size, loading time.. 🤔

File: proposals/mui-v5-migration.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+# Plan
+
+Since there are many changes waiting to be migrated, it will be quite logical to have this migration gradually. So, initial plan is to establish two root versions of Picasso component that enable using MUI v4 and v5 working at the same time until everything settles down. To achieve that, we need these options to be handled first:
```
Comment (dmaklygin) [inline proposals/mui-v5-migration.md:15]: you mean one `Picasso component` or all Picasso components?

File: proposals/mui-v5-migration.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+- MUIv5 theme structure also changed. So, there may be a new set of theme types and necessary other files created like typograpy, shadows and breakpoints.
+- MUIv5 has updates about overriding existing MUI styles. So we need to clone our existing Picasso component and tweak it to work with v5 overrides. We will use this new version of Picasso component in our new v5 root.
+- Update both storybook and visual test code to use different root for different components. This will give us the chance to migrate one component and check the visual changes without touching and breaking anything around the repo.
+
+After accomplishing these points, we can be ready to migrate individual core components of MUI inside Picasso with the power of using existing tests. This plan will also avoid the requirement for a feature branch. We can directly merge one component's migration PR to master without waiting or handling conflicts for a possible feature branch.
+
+So, preparing a PoC with minimal changes on these points and a sample component would be a good idea.
+
```
Comment (denieler) [inline proposals/mui-v5-migration.md:24]: The final step will be to remove v4 root Picasso and replace it with v5 root Picasso

File: proposals/mui-v5-migration.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+As a core foundation, we are using MUI for several reasons like theming and JSS solutions. So, it would be good to migrate it to version 5 to be able to have more reliable and supported base with better CSS-in-JS performance.
+
+# Goal
+
+The goal of this migration is to have fully migrated core components of MUI from version 5 and emotion based CSS-in-JS solution with minimal number of breaking changes reflected to Picasso clients. It also means that we shouldn't be reflecting anything related to MUI did internally for meeting their own design purposes.
+
+To be able to make things straight forward, it would be good to postpone removal of all JSS parts (like @mui/styles package usages) from the repo before everthing is working fine again. So, `styled` or `TSS-react` solutions can be discussed in further iterations.
+
+# Plan
```
Comment (denieler) [inline proposals/mui-v5-migration.md:13]: in this plan we would be able to use MUI 5 codemods still 🤔

File: proposals/mui-v5-migration.md (hunk 3 above)
Comment (ozgurkececioglu) [reply proposals/mui-v5-migration.md:15]: There is a typo, I was refering to `two versions of Picasso root component`. And it was the component itself: `packages/picasso-provider/src/Picasso/Picasso.tsx`

File: docs/decisions/mui-v5-migration.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+Migrate Picasso to use MUI v5 with minimal breaking changes
+
+## Plan
+
+Since migration requires many changes, we want to do it gradually.
+
+- create feature branch
+- create new version of `Picasso` root component (e.g: `PicassoV5`) for new styled engine, theming and style overrides
+- configure storybook to use `PicassoV5` with migrated components while running storybook visual tests
```
Comment (OleksandrNechai) [inline docs/decisions/mui-v5-migration.md:18]: I guess this includes to only use `PicassoV5` for components that are migrated so we guarantee that feature branch is always green 🟢

File: docs/decisions/mui-v5-migration.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+- migrate all components
+- replace old Picasso root with `PicassoV5`
+- remove MUIv4 packages
+- migrate Picasso to use tss-react
```
Comment (denieler) [inline docs/decisions/mui-v5-migration.md:22]: or styled 😃

File: docs/decisions/mui-v5-migration.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+_Pros:_
+
+- `emotion` is very similar to our current `makeStyles` CSS solution, so fewer changes
+
+_Cons:_
+
+- none
+
+**We propose to use `emotion`**
```
Comment (denieler) [inline docs/decisions/mui-v5-migration.md:38]: 👍

**Review by TomasSlama — APPROVED** [review]
Nice document 👍

**Review by dmaklygin — APPROVED** [review]

**Review by OndrejTuma — APPROVED** [review]

**Pull request merged**